### ISSUE TITLE
get_special_tokens_mask doesn't appear to prevent [cls] and the like from being masked

### ISSUE THREAD

**Issue by cowsock** [matejMartinc/scalable_semantic_shift#2]
Hi! I'm a new PhD student working in the area of lexical semantic change, and just getting started with my own implementations / experiments / etc. (I say this just to humble myself before describing this issue!) Thanks!

https://github.com/matejMartinc/scalable_semantic_shift/blob/a105c8409db0996c99f0df11d40c35017eb3337c/fine-tune_BERT.py#L449-L451

looks like it will never return any `1`s or `True`s even when the input is `CLS` or a padding token. Is this intentional?

Alternative implementations would be something like from https://github.com/huggingface/transformers/blob/7ec1dc8817a99d16e6f9e0ab94ce4027ef74b72d/src/transformers/data/data_collator.py#L742-L744

```
special_tokens_mask = [
                self.tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True) for val in labels.tolist()
]
special_tokens_mask = torch.tensor(special_tokens_mask, dtype=torch.bool)
```

or (rephrased by me):

```
special_tokens_mask = [self.get_special_tokens_mask(val) for val in labels.tolist()]
...
def get_special_tokens_mask(self, token_ids_0):
        return [1 if ident in self.tokenizer.all_special_ids else 0 for ident in token_ids_0]
```

**Comment by matejMartinc** [comment]
Hi! This is in fact a somewhat old code from the Huggingface examples for model fine-tuning. I'm not sure about  the get_special_tokens_mask function functionality, but I do suggest you use the latest Huggingface code for model fine-tuning for best results.
